src/Microsoft.DotNet.Darc/src/DarcLib/Models/Darc/DependencyGraph.cs

```diff
@@ -143,14 +190,181 @@ public class DependencyGraph
+                            latestCommitCache.Add(latestCommitKey, latestCommit);
+                        }
+
+                        // Perform diff if there is a latest commit.
+                        if (!string.IsNullOrEmpty(latestCommit))
+                        {
+                            IRemote repoRemote = remoteFactory.GetRemote(node.RepoUri, logger);
+                            // This will return a no-diff if latestCommit == node.Commit
+                            node.DiffFrom = await repoRemote.GitDiffAsync(node.RepoUri, latestCommit, node.Commit);
+                        }
+                    }
+                }
+            }
+        }
+
+        /// <summary>
+        ///     Diff each node in the graph against the latest build in
+        ///     the graph.
+        /// </summary>
+        /// <param name="remoteFactory"></param>
+        /// <param name="logger"></param>
+        /// <param name="nodeCache"></param>
+        /// <param name="visitedRepoUriNodes"></param>
+        /// <returns></returns>
+        private static async Task DoLatestInGraphNodeDiffAsync(
+            IRemoteFactory remoteFactory,
+            ILogger logger,
+            Dictionary<string, DependencyGraphNode> nodeCache,
+            Dictionary<string, DependencyGraphNode> visitedRepoUriNodes)
+        {
+            logger.LogInformation("Running latest in graph node diff.");
+
+            // Find the build of each repo in the graph, then
+            // get the diff info from the latest
+            Dictionary<string, DependencyGraphNode>.KeyCollection uniqueRepos = visitedRepoUriNodes.Keys;
+            foreach (string repo in uniqueRepos)
+            {
+                // Get all nodes with this value
+                IEnumerable<DependencyGraphNode> nodes = nodeCache.Values.Where(n => n.RepoUri == repo);
+                // If only one, determine latest
+                if (nodes.Count() > 1)
+                {
+                    // Find latest
+                    DependencyGraphNode newestNode = null;
+                    Build newestBuild = null;
+                    foreach (DependencyGraphNode node in nodes)
+                    {
+                        if (newestNode == null)
+                        {
+                            newestNode = node;
+                            if (newestNode.ContributingBuilds.Any())
+                            {
+                                newestBuild = newestNode.ContributingBuilds.OrderByDescending(b => b.DateProduced).First();
+                            }
+                        }
+                        else if (node.ContributingBuilds.Any())
+                        {
+                            // Determine whether any of the builds are newer
+                            if (node.ContributingBuilds.Any(b => newestBuild == null ||
+                                                            b.DateProduced.Value.CompareTo(newestBuild.DateProduced.Value) > 0))
```

Comment:
>b.DateProduced.Value [](start = 60, length = 20)

What is this object that we can't just do `b.DateProduces.Value > newestBuild?.DateProduceds.Value`, rather than the null and then the compare to? Based on the names, they are just dates, which should compare directly fine.